cmd/argocd/commands/app.go

```diff
@@ -162,6 +165,7 @@ func NewApplicationCreateCommand(clientOpts *argocdclient.ClientOptions) *cobra.
 	command.Flags().StringVarP(&fileURL, "file", "f", "", "Filename or URL to Kubernetes manifests for the app")
 	command.Flags().StringArrayVarP(&labels, "label", "l", []string{}, "Labels to apply to the app")
 	command.Flags().StringArrayVarP(&annotations, "annotations", "", []string{}, "Set metadata annotations (e.g. example=value)")
+	command.Flags().BoolVar(&setFinalizer, "set-finalizer", false, "Sets deletion finalizer on the application, application resources will be cascaded on deletion")
```

Comment:
```suggestion
	command.Flags().BoolVar(&setFinalizer, "set-finalizer", false, "Sets deletion finalizer on the application")
```
I think it's sufficient to mention the finalizer. The second part wouldn't hold true if the app is deleted using the API (e.g. CLI or API).

I was wondering, `argocd app delete` has the `--cascade` flag, which would set the finalizer. I have not tested, but how does the deletion process behave when the finalizer set is set on the app, and `argocd app delete` would be called _without_ `--cascade` flag?

Reply:
I think "deletion is cascaded" will always be true if finalizer is set, even  if app is deleted with cli or ui. bcoz there is no feature to remove finalizer while deletion.

>I was wondering, argocd app delete has the --cascade flag, which would set the finalizer. I have not tested, but how does the deletion process behave when the finalizer set is set on the app, and argocd app delete would be called without --cascade flag?

if --cascade=true it checks ```resources-finalizer.argocd.argoproj.io finalizer``` is present or not, if not present it patches application to add finalizer,  else it skips and just deletes application.

if during creation of application, finalizer is set , even if we set --cascaded=false  while deleting application using cli, all resources managed by application will be deleted.

Reply:
Hence i am in favour to keep message that deletion will be cascaded

Reply:
@jannfis Please share thoughts on how the text should be? merge suggestion or not?

Reply:
Question @rishabh625 / @jannfis , I am wondering what the behavior should be post-app creation with the set finalizer option checked.  After the app has been created, and you decide to delete one of the resources from the UI, the delete dialog appears with the foreground delete option selected (the non-cascading delete is NOT selected).  What should happen if the user chooses the non-cascading delete option?  The UI will call the application service with the cascade option set to false.  ( https://github.com/argoproj/argo-cd/blob/3659b7bb7d7c4990ee8f5d55ec1a816fa78e5aff/ui/src/app/shared/services/applications-service.ts#L120 )

So should the UI be 'smart' to know that there is a finalizer, and perhaps this option should be disabled in the UI?  (The user might expect the resources to be orphaned instead).

In terms of the create panel changes in the UI, it looks good now and the variable names look ok.  The state of the newly added checkbox is synced up with whatever is shown in the `Edit As Yaml` page. In terms of these specific changes, it LGTM.

Reply:
So that's the purpose of this PR, that this behaviour to delete cascading or not, we need to set at creation of resource,  and not while deletion, we know that cr can be deleted with several way with argocd ui/cli or with kube api

When deleted with kube api we don't have control on this behaviour,so with creation of application resource with this optional flag we add this behaviour

Concern that we should disable non cascading delete if cascading behaviour was set during creation should be done but that would be difficult I guess.

If when delete is clicked and we have this finalizer we can prompt that this will be deleted cascading

Reply:
I thought it as simple change, will wait for @jannfis  to suggest